File: javascript/plusOne.js
/**
 * @param {number[]} digits
 * @return {number[]}
 */
var plusOne = function(digits) {
    // let num = BigInt(digits.join(""))
    // num++
    // return num.toString().split('')
    
    let n = digits.length - 1
    console.log(n)
    
    
    
    function checker (position, nums){
        if (position == 0 && nums[position] == 9) {
            nums[0] = 1
            nums.push(0)
            return nums
        }
        else if (nums[position] != 9){
            nums[position] += 1
            return nums
        }
        else if(nums[position] == 9 && position != 0){
            nums[position] = 0 
            position = position - 1
    
            return checker(position, nums)
        }
    }
    
    return checker(n, digits)
    
    
    //  let n = digits.length - 1;
    //     for(let i=n; i>=0; i--){
    //         if(digits[i] !== 9){
    //             digits[i]++;
    //             break;
    //         } else {
    //             digits[i] = 0;
    //             if(i===0){
    //                 digits.unshift(1);
    //             }
    //         }
    //     }
    //     return digits;
    };

console.log(plusOne([9]));
console.log(plusOne([9,9]));
console.log(plusOne([1,2,3]));
console.log(plusOne([2,4,6,7,9,9,3,4]));

File: python/plusone.py
def check(n,lists):
    if n == 0 and lists[n] ==9:
        lists[0] = 1
        lists.append(0) 
        return lists

    elif  lists[n] != 9  :
        value = lists[n]
      
        lists[n]  = value + 1 

        return lists


    elif lists[n] == 9  and n !=0:
        lists[n],n = 0, n-1

        return check(n , lists)
   

class Solution:
    def plusOne(self, digits):
      n  = len(digits)-1 
      
      return check(n,digits)


plusOne = Solution()

print(plusOne.plusOne([9,9]))
print(plusOne.plusOne([9,9,9]))
print(plusOne.plusOne([9,9,2,6,9,3,5,2,7,8,6]))
